contracts/zaps/DepositZapMorphoAaveV2USD.vy

```diff
@@ -0,0 +1,279 @@
+
+    # Deposit all coins into vaults
+    for i in range(BASE_N_COINS):
+        amount: uint256 = _deposit_amounts[i]
+        if amount == 0:
+            continue
+
+        coin: address = base_coins[i]
+        ma_coin: address = ma_coins[i]
+
+        ERC20(coin).transferFrom(msg.sender, self, amount)
+        shares[i] = SupplyVault(ma_coin).deposit(amount, self)
+
+    # Deposit to the base pool
+    return CurveBase(BASE_POOL).add_liquidity(shares, _min_mint_amount, _receiver)
+
+
+@external
+def remove_liquidity(
+    _pool: address, # TODO: should we use this in place of BASE_POOL?
+    _burn_amount: uint256,
+    _min_amounts: uint256[BASE_N_COINS],
+    _receiver: address = msg.sender
+) -> uint256[BASE_N_COINS]:
+    """
+    @notice Withdraw and unwrap coins from the pool
+    @dev Withdrawal amounts are based on current deposit ratios
+    @param _pool Address of the pool to deposit into
+    @param _burn_amount Quantity of LP tokens to burn in the withdrawal
+    @param _min_amounts Minimum amounts of underlying coins to receive
+    @param _receiver Address that receives the LP tokens
+    @return List of amounts of underlying coins that were withdrawn
+    """
+    ERC20(BASE_POOL).transferFrom(msg.sender, self, _burn_amount)
+
+    min_shares: uint256[BASE_N_COINS] = self._calc_shares(_min_amounts, False)
+
+    # Withdraw from base
+    base_shares: uint256[BASE_N_COINS] = CurveBase(BASE_POOL).remove_liquidity(_burn_amount, min_shares)
+
+    # Withdraw all coins from vaults
+    ma_coins: address[BASE_N_COINS] = MA_COINS
+    amounts: uint256[BASE_N_COINS] = empty(uint256[BASE_N_COINS])
+    for i in range(BASE_N_COINS):
+        shares: uint256 = base_shares[i]
+        if shares == 0:
+            continue
+
+        ma_coin: address = ma_coins[i]
+
+        amounts[i] = SupplyVault(ma_coin).redeem(shares, _receiver, self)
+
+    return amounts
+
+
+@external
+def remove_liquidity_one_coin(
+    _pool: address, # TODO: should we use this in place of BASE_POOL?
+    _burn_amount: uint256,
+    i: int128,
+    _min_amount: uint256,
+    _receiver: address=msg.sender
+) -> uint256:
+    """
+    @notice Withdraw and unwrap a single coin from the pool
+    @param _pool Address of the pool to deposit into
+    @param _burn_amount Amount of LP tokens to burn in the withdrawal
+    @param i Index value of the coin to withdraw
+    @param _min_amount Minimum amount of underlying coin to receive
+    @param _receiver Address that receives the LP tokens
+    @return Amount of underlying coin received
+    """
+    ERC20(BASE_POOL).transferFrom(msg.sender, self, _burn_amount)
+
+    ma_coins: address[BASE_N_COINS] = MA_COINS
+    ma_coin: address = ma_coins[i]
+
+    min_shares: uint256 = SupplyVault(ma_coin).previewWithdraw(_min_amount)
+
+    # Withdraw a base pool coin
+    shares: uint256 = CurveBase(BASE_POOL).remove_liquidity_one_coin(_burn_amount, i, min_shares) # TODO: does not support fee on transfer
```

Comment:
Should we pass `min_shares` or `min_amount` instead?

Reply:
`min_shares` because the zap is withdrawing from the pool of maTokens (and amount is in assets)